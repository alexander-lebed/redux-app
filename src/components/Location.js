// @flow
import React from 'react';
import {Map} from 'immutable';
import Col from 'react-bootstrap/lib/Col';
import Row from 'react-bootstrap/lib/Row';
import Glyphicon from 'react-bootstrap/lib/Glyphicon';

type Props = {
    location: Map<string, any>,
    onDelete: (id: string) => void
}


const Location = (props: Props) => {
    const {location, onDelete} = props;
    let temperature = '';
    if (location.get('temp')) {
        const degreesInCelsius = Math.round((location.get('temp') - 32) / 1.8);
        temperature = `${degreesInCelsius} Â°C`;
    } else {
        temperature = 'not available';
    }
    return (
        <Row>
            <Col xs={6}>{location.get('city')}</Col>
            <Col xs={3}>{temperature}</Col>
            <Col xs={3}>
                <Glyphicon
                    glyph="remove"
                    className="pull-right"
                    style={{marginRight: 10}}
                    title="Remove Location"
                    onClick={onDelete}
                />
            </Col>
        </Row>
    );
};

export default Location;