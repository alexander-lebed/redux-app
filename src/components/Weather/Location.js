// @flow
import React from 'react';
import { Row, Col, Glyphicon } from 'react-bootstrap';
import { MAIN_COLOR } from '../../constants';
import type { Location as LocationType } from '../../types';


type Props = {
    location: LocationType,
    onDelete: (id: string) => void
}

export const Location = (props: Props) => {
    const { location, onDelete } = props;
    let tempInfo = '';
    if (location.temp) {
        const degreesInCelsius = Math.round((location.temp - 32) / 1.8);
        tempInfo = <span><strong>{degreesInCelsius}</strong>Â°C</span>;
    } else {
        tempInfo = 'not available';
    }
    return (
        <Row style={{paddingTop: 5, paddingBottom: 5}}>
            <Col xs={6}>
                <Glyphicon glyph='map-marker' style={{marginRight: 5, color: MAIN_COLOR}} />
                {`${location.city}${location.region && ` (${location.region})`}, ${location.country}`}
            </Col>
            <Col xs={4}>{tempInfo}{location.text && `, ${location.text.toLowerCase()}`}</Col>
            <Col xs={2}>
                <Glyphicon
                    glyph="trash"
                    className="pull-right"
                    style={{marginRight: 10, color: 'grey'}}
                    title="Remove Location"
                    onClick={onDelete}
                />
            </Col>
        </Row>
    );
};

export default Location;
