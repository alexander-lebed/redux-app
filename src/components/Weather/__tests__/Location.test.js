// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Location from '../Location';


describe('<Location/>', () => {
    test('should render Location', () => {
        const location = {id: 'Odessa(OD),Ukraine', city: 'Odessa', region: 'OD', country: 'Ukraine', temp: 0, isDeleted: false};
        const wrapper = renderer.create(
            <Location location={location} onDelete={jest.fn()} />
        );
        const tree = wrapper.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

