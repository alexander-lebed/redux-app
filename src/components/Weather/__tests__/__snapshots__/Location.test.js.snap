// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Location/> should render Location 1`] = `
<div
  className="row"
  style={
    Object {
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <div
    className="col-xs-4"
  >
    <span
      className="glyphicon glyphicon-map-marker"
      style={
        Object {
          "color": "#00b386",
          "marginRight": 5,
        }
      }
    />
    Odessa
  </div>
  <div
    className="col-xs-6"
  >
    <span>
      <strong>
        27
      </strong>
      Â°C
    </span>
  </div>
  <div
    className="col-xs-2"
  >
    <span
      className="pull-right glyphicon glyphicon-remove"
      onClick={[Function]}
      style={
        Object {
          "color": "grey",
          "marginRight": 10,
        }
      }
      title="Remove Location"
    />
  </div>
</div>
`;
